src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -331,30 +332,32 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
 
             // expand not 'safe' characters into %XX, spaces to +s
             byte[] expandedBytes = new byte[count + cUnsafe * 2];
-            int pos = 0;
+            GetEncodedBytes(bytes, 0, expandedBytes);
```

Comment:
good catch. I'm surrprised there are no tests for this. I'll fix this and add a test